samcli/commands/local/lib/sam_api_provider.py

```diff
@@ -91,65 +94,16 @@ def _extract_apis(self, resources):
         # corresponding Serverless::Api resource. This is all done using the ``collector``.
 
         collector = ApiCollector()
-
+        providers = {Provider.RESOURCE_TYPE: Provider() for Provider in AbstractParserProvider.__subclasses__() if
```

Comment:
I am having a hard time understand the intention here. Can you explain what you are trying to accomplish? It looks like you are trying to create a dict of a Resource type to the provider. Why not statically define this?